src/drive/web/modules/navigation/FileExplorer.jsx

```diff
@@ -88,7 +92,22 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
   fetchRecentFiles: () => dispatch(fetchRecentFiles()),
   fetchMoreFiles: (folderId, skip, limit) =>
     dispatch(fetchMoreFiles(folderId, skip, limit)),
-  fetchFolder: folderId => dispatch(openFolder(folderId)),
+  fetchFolder: folderId => {
+    // TODO: this should check the hierarchy
+    if (folderId === VAULT_DIR_ID) {
+      dispatch(
+        showModal(
+          <Passphrase
+            onSubmitPassphrase={() => {
```

Comment:
Why setting a function as a `prop` of componnt `Passphrase` since the `onSubmitPassphrase` is already a method of this component ?

Reply:
As you see, `onSubmitPassphrase` does nothing here, it is an anticipation for the next steps: get callback from the `Passphrase` component to do stuff, e.g. open the folder only if the entered passphrase is ok.
The  `onSubmitPassphrase` method from Passphrase is its child counterpart and could (should?) actually be named differently. I actually didn't notice the same names (the child one came after, I originally thought the parent was enough), so no problem to change this

Reply:
OK for a renaming, then :)